code/API_definitions/predictive-connectivity-data-vwip.yaml

```
@@ -0,0 +1,1045 @@
+          - `OPERATION_NOT_COMPLETED`: An error happened during asynchronous processing of the request. This status will only be returned
+          in case the asynchronous API behaviour is used.
+      enum:
+        - SUPPORTED_AREA
+        - PART_OF_AREA_NOT_SUPPORTED
+        - AREA_NOT_SUPPORTED
+        - OPERATION_NOT_COMPLETED
+    TimedConnectivityData:
+      type: object
+      description: >-
+        Represents a time interval with connectivity data for the cells in the
+        area. Time interval length is
+        determined by the implementation. The request startTime or the request
+        endTime have to be fully covered by the intervals. For example, if the
+        intervals are 1-hour long and the input date range were
+        [2024-01-03T11:25:00Z to 2024-01-03T12:45:00Z] it would contain 2
+        intervals (Interval from 2024-01-03T11:00:00Z to 2024-01-03T12:00:00Z
+        and interval from 2024-01-03T12:00:00Z to 2024-01-03T13:00:00Z).
+      properties:
+        startTime:
+          type: string
+          format: date-time
+          description: >-
+            Interval start time. It must follow [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6)
+            and must have time zone. Recommended format is yyyy-MM-dd'T'HH:mm:ss.SSSZ
+            (i.e. which allows 2023-07-03T14:27:08.312+02:00 or 2023-07-03T12:27:08.312Z)
+          example: "2023-07-03T10:00:00Z"
+        endTime:
+          type: string
+          format: date-time
+          description: >-
+            Interval end time. It must follow [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6)
+            and must have time zone. Recommended format is yyyy-MM-dd'T'HH:mm:ss.SSSZ
+            (i.e. which allows 2023-07-03T14:27:08.312+02:00 or 2023-07-03T12:27:08.312Z)
+          example: "2023-07-03T11:00:00Z"
+        cellConnectivityData:
+          $ref: '#/components/schemas/CellConnectivityDataArray'
+      required:
+        - startTime
+        - endTime
+        - cellConnectivityData
+    CellConnectivityDataArray:
+      type: array
+      description: >-
+        Connectivity data for the different cells in a concrete time range.
+      items:
+        $ref: '#/components/schemas/CellConnectivityData'
+      minItems: 1
+    CellConnectivityData:
+      description: >-
+        Represents a rectangular grid cell within the specified area, including
+        its connectivity data.
+      properties:
+        geohash:
+          type: string
+          description: >-
+            Coordinates of the cell represented as a string using the [Geohash
+            system](https://en.wikipedia.org/wiki/Geohash), encoding a
+            geographic location into a short string. The value length, and thus
+            the cell granularity, is determined by the implementation.
+          example: ezdmemd
+        layerConnectivities:
+          $ref: '#/components/schemas/LayerConnectivities'

```

Add option to include signal strength estimates in response
```suggestion
          $ref: '#/components/schemas/LayerConnectivities'
        signalStrength:
          type: array
          items:
            description: Predicted signal strength values in dBm for each layer
            type: number
```

---

@VGF-TEF 
Thanks Eric. From Telefónica’s perspective, we currently do not provide signal strength prediction as a service level, so we would not be able to support this value at the moment.

If SIGNAL_STRENGTH is defined as an optional service level that providers can choose whether or not to implement, we are fine with its inclusion. However, if support becomes mandatory for all providers, we would need to push back.